tests/integration/test_integration_all_rust.py

```diff
@@ -243,52 +240,40 @@ def print_lines_in_queues(queues, prefix):
                 sys.stdout.write(prefix + line)
 
 
-def process_logs(testcase):
-    """Process (print) the testcase logs (in tearDown).
+@pytest.fixture(name="process_logs_autouse", autouse=True, scope="function")
+def fixture_process_logs_autouse():
```

Comment:
Does this need to be a fixture?

I'll argue the decorator version of this is nicer since it didn't require the test to call `process_logs` themselves. Granted the previous decorator accomplished that via `process_logs` in the `tearDown` hook which doesn't exist any longer, but I think if this was still a decorator it could handle that equivalent of `process_logs` by itself?

If so I think that further suggests it shouldn't be a fixture as tests wouldn't require any parameter/object provided from it (such as `process_logs_autouse`).

Reply:
Thanks again @pjenvey , as we discussed offline it's now using the parametrized fixtures that pass in the context much the same way with the decorator, however avoiding the async funkiness... 

Two named fixtures that conditionally return an int repr of that given log value:
```python
@pytest.fixture
def fixture_max_conn_logs(request):
    """Fixture returning altered max_conn_logs value for process_logs fixture."""
    if hasattr(request, "param"):
        return int(request.param)


@pytest.fixture
def fixture_max_endpoint_logs(request):
    """Fixture returning altered max_endpoint_logs value for process_logs fixture."""
    if hasattr(request, "param"):
        return int(request.param)
```
        
Single param invocation:

```python
@pytest.mark.parametrize("fixture_max_conn_logs", [4], indirect=True)
```

Multi param invocation
```python
@pytest.mark.parametrize(
    "fixture_max_endpoint_logs, fixture_max_conn_logs",
    [[4, 1]],
    indirect=["fixture_max_endpoint_logs", "fixture_max_conn_logs"],
)
```